lib/rspec/rails/matchers/have_enqueued_mail.rb

```diff
@@ -88,6 +88,18 @@ def arguments_match?(job)
           super(job)
         end
 
+        def process_arguments(job, given_mail_args)
+          if job[:job] == ActionMailer::MailDeliveryJob
+            if given_mail_args.first.is_a?(Hash) && job[:args][3]['params'].present?
+              [hash_including(params: given_mail_args[0], args: given_mail_args.drop(1))]
+            else
+              [hash_including(args: given_mail_args)]
+            end
+          else
```

Comment:
What the `job[:job]` is in this case?

Reply:
Any other delivery job that's not `ActionMailer::MailDeliveryJob`, if it behaves like the existing one, then current code is fine, else the matcher would be broken also without this PR.